libqtile/bar.py

```diff
@@ -222,8 +222,12 @@ def _configure(self, qtile, screen):
         qtile.windows_map[self.window.window.wid] = self.window
         self.window.unhide()
 
-        for i in self.widgets:
-            qtile.register_widget(i)
+        for idx, i in enumerate(self.widgets):
+            if i.configured:
+                i = i.create_mirror()
+                self.widgets[idx] = i
+            else:
+                qtile.register_widget(i)
```

Comment:
So it looks like we skip registering the widget, which isn't ideal. For example, we call widgets' finalize() methods based on there presence in widgets_map, which means we'll leak X resources if we don't include them here. Can we include them, maybe with "-mirror" tacked on or something?

Reply:
Good point - always registering and removing the `else` should do it. Looking at the register_widget code though I'm not sure it is registering all widgets every time. If you mirror a widget more than once then the repeated name stops all others being registered. Could qtile.widgets_map not just be a list of all registered widgets instead of a dict using their names?

Reply:
Yeah, I think the only reason it's a list now is that everything was unique.

In fact you're right, if people have two bars with a GroupBox() on either of them, it will only register the first one. So we're already leaking this anyway. Switching it to a list would be good.

Reply:
So to resolve this, can we just call .register_widget() on everything, and later we can fix that code to use a map instead of a list?

Reply:
Sure!